pkg/chaosdaemon/tasks/task_manager.go

```diff
@@ -0,0 +1,278 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package tasks
+
+import (
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+
+	"github.com/chaos-mesh/chaos-mesh/pkg/chaoserr"
+)
+
+var ErrCanNotAdd = errors.New("can not add")
+var ErrCanNotAssign = errors.New("can not assign")
+
+// Injectable stand for the base behavior of task : inject a process with PID.
+type Injectable interface {
+	Inject(pid PID) error
+}
+
+// Recoverable introduce the task recovering ability.
+// Used in Recover.
+type Recoverable interface {
+	Recover(pid PID) error
+}
+
+// Creator init an Injectable with values.
+// We use it in a case that TaskConfig.data init an Injectable task here.
+type Creator interface {
+	New(values interface{}) (Injectable, error)
+}
+
+// Assign change some of an Injectable task with its own values.
+// We use it in a case that we use TaskConfig.data
+// to update an Injectable task.
+type Assign interface {
+	Assign(Injectable) error
+}
+
+// TaskExecutor indicate that the type can be used
+// for execute task here as a task config.
+// Addable means we can sum many task config in to one for apply.
+// Creator means we can use the task config to create a running task
+// which can Inject on PID.
+// Assign means we can use the task config to update an existing running task.
+type TaskExecutor interface {
+	Addable
+	Creator
+	Assign
+}
+
+// TaskManager is a Manager for chaos tasks.
+// A task base on a target marked with its PID.
+// We assume task should implement Injectable.
+// We use TaskConfig.data which implement TaskExecutor to:
+//	Sum task configs on same PID into one.
+// 	Create task.
+// 	Assign or update task.
+// SO if developers wants to use functions in TaskManager ,
+// their imported TaskConfig need to implement interface TaskExecutor.
+// If developers wants to recover task successfully,
+// the task must implement Recoverable.
+// If not implement ,
+// the Recover function will return a ErrNotImplement("Recoverable") error.
+// IMPORTANT: We assume task config obey that TaskConfig.data A,B. A.Add(B)
+// is approximately equal to B.Add(A)
+type TaskManager struct {
+	taskConfigManager TaskConfigManager
+	taskMap           map[PID]Injectable
+
+	logger logr.Logger
+}
+
+func NewTaskManager(logger logr.Logger) TaskManager {
+	return TaskManager{
+		NewTaskConfigManager(),
+		make(map[PID]Injectable),
+		logger,
+	}
+}
+
+func (cm TaskManager) CopyTaskConfigManager() TaskConfigManager {
+	tm := NewTaskConfigManager()
+	for uid, task := range cm.taskConfigManager.TaskConfigMap {
+		tm.TaskConfigMap[uid] = task
+	}
+	return tm
+}
+
+func (cm TaskManager) CopyTaskMap() map[PID]Injectable {
+	pm := make(map[PID]Injectable)
+	for pid, chaosOnProcess := range cm.taskMap {
+		cm.taskMap[pid] = chaosOnProcess
+	}
+	return pm
+}
+
+func (cm TaskManager) GetConfigWithUID(id UID) (interface{}, error) {
```

Comment:
Are functions `GetConfigWithUID` and `GetTaskWithPID` only used in the tests?

Reply:
Yse, and I want to ensure developer have the ability to get all the entity inside my package.